### PR TITLE
Add training loop to docs

### PR DESCRIPTION
**Pull request by DhairyaLGandhi**
This also links to the correct point in the docs to read about custom training loop behaviour.

### CODE REVIEW

**Comment by DhairyaLGandhi** [comment]
cc @ToucheSir

**Review by ToucheSir — APPROVED** [review]
Looks great, thanks for adding this!

File: docs/src/training/training.md [hunk 1]
```diff
@@ -7,13 +7,36 @@ To actually train a model we need four things:
...
+```
+
+To make it easy, Flux defines `train!`:
 
 ```@docs
 Flux.Optimise.train!
 ```
 
-There are plenty of examples in the [model zoo](https://github.com/FluxML/model-zoo).
+There are plenty of examples in the [model zoo](https://github.com/FluxML/model-zoo), and
```
Comment (darsnack) [inline docs/src/training/training.md:38]:
```suggestion
Like the loop above, `train!` expects the data to be [iterable](https://docs.julialang.org/en/v1/manual/interfaces/#man-interface-iteration). Most default collections Julia, like `Array`s, are iterable.

There are plenty more examples in the [model zoo](https://github.com/FluxML/model-zoo), and
```

Comment (DhairyaLGandhi) [reply docs/src/training/training.md:38]: We already mention and demonstrate the iteration twice in the couple lines above, do we need it again here?

Comment (darsnack) [reply docs/src/training/training.md:38]: People unfamiliar to Julia may not realize that the syntax `for x in xs` requires the iteration interface. The suggestion was made to make understanding the resulting method error easier for new users.

**Pull request merged**